_posts/2018-03-23-crossword-data-format.md

```diff
@@ -0,0 +1,116 @@
+
+and drives [an interactive version](https://labs.ft.com/crosswords/2018/03/07/Puzzle-15800-dynamic.html).
+
+# The new format
+
+The crossword data format is a DSL, [a Domain Specific Language](https://en.wikipedia.org/wiki/Domain-specific_language), being an concise, clear, human-readable piece of text that specifies how to construct a simple crossword.
+
+It is written in [YAML](https://en.wikipedia.org/wiki/YAML). Or rather, nearly YAML. The line items for each clue need to be wrapped in quotes in order to accommodate colons and speech marks in the clue. Currently our implementation fails to take this into account, parsing the text directly and not as YAML.
+
+Blank lines and comments (starting with #) are ignored.
+
+There are several mandatory fields such as ‘name’ (which should be unique across all the crosswords), ‘author’, and ‘size’.
+
+The clues are grouped as lists named ‘across’ and ‘down’.
+
+Each clue has the form: - `(COORDINATES) IDS. Clue text (ANSWER)`
+
+COORDINATES are (across, down), with (1,1) being the top left square.
+
+The IDS identify each clue, and also tie together subsequent (comma-separated) clue ids for answers which straddle multiple positions on the grid. The subsequent IDS numbers might have an associated ‘down’ or ‘across’ if needed to disambiguate when a particular clue id has both aspects.
+
+The ANSWER is a sequence of words or numbers separated by commas or hyphens. Very obscure crosswords might want to have full stops, colons, or semi-colons in there too, and this format could handle them.
+
+The Clue text can include basic html formatting such as italics. A restriction of the YAML format means that there can be no bare square brackets, speech marks, colons, etc in the clue and answer texts, but the line (after the initial hyphen) can be wrapped in quotes to overcome that.
+
+# Using the format
+
+The format is easy to read, debug, and edit manually. This is probably its main benefit when compared with the other formats out there.
+
+It is easy to parse and validate, at a basic level, to check that all the mandatory details and structure are present and correct.
+
+A little more effort (by unpacking the clues into a grid) is needed to ensure that
+
+* the answers fit within the stated size of the grid,
+* the clues are numbered appropriately,
+* and the answer letters agree at every intersection .
+
+We have several different incoming data formats for crosswords published by the FT, including some (undocumented) XML exported from a commercial crossword compiling tool, [CCW](https://www.crossword-compiler.com/), and a bespoke (but consistently-formatted) word doc. It has proved relatively simple to transform them into the new YAML format, and import that into our new crossword implementation.
+
+# Weaknesses
```

Comment:
I would put this section as a subsection of `The new format`

Reply:
done